src/clue.js

```diff
@@ -2,25 +2,145 @@
+        age: 22,
+        description: 'Billionaire video game designer',
+        image: 'https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg',
+        color: 'purple',
+    },
+    {
+        firstName: 'Kasandra',
+        lastName: 'Scarlet',
+        occupation: 'Actor',
+        age: 31,
+        description: 'She is an A-list movie star with a dark past',
+        image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+        color: 'red',
+    },
+    {
+        firstName: 'Eleanor',
+        lastName: 'Peacock',
+        occupation: 'Socialité',
+        age: 36,
+        description: 'She is from a wealthy family and uses her status and money to earn popularity',
+        image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+        color: 'blue',
+    },
+    {
+        firstName: 'Jack',
+        lastName: 'Mustard',
+        occupation: 'Retired Football player',
+        age: 62,
+        description: 'He is a former football player who tries to get by on his former glory',
+        image: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg',
+        color: 'yellow',
+    }
+];
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [
+    {
+        name: 'rope',
+        weight: 10,
+    },
+    {
+        name: 'knife',
+        weight: 8,
+    },
+    {
+        name: 'caddlestick',
+        weight: 2,
+    },
+    {
+        name: 'dumbbell',
+        weight: 30,
+    },
+    {
+        name: 'poison',
+        weight: 2,
+    },
+    {
+        name: 'axe',
+        weight: 15,
+    },
+    {
+        name: 'bat',
+        weight: 13,
+    },
+    {
+        name: 'trophy',
+        weight: 25,
+    },
+    {
+        name: 'pistol',
+        weight: 20,
+    }
+];
 
+// Rooms Array
+
+const roomsArray = [
+    
+    {name: 'Dining Room'},
+    {name: 'Conservatory'},
+    {name: 'Kitchen'},
+    {name: 'Study'},
+    {name: 'Library'},
+    {name: 'Billiard Room'},
+    {name: 'Lounge'},
+    {name: 'Ballroom'},
+    {name: 'Hall'},
+    {name: 'Spa'},
+    {name: 'Living Room'},
+    {name: 'Observatory'},
+    {name: 'Theater'},
+    {name: 'Guest House'},
+    {name: 'Patio'},
+]
 
 // ITERATION 2
 
-function selectRandom() {}
+function selectRandom(array) {
+    let randomIndex = Math.floor(Math.random()* array.length);
+    return array[randomIndex]
+}
 
-function pickMystery() {}
+function pickMystery() {
+    let suspect = selectRandom(suspectsArray)
+    let weapon = selectRandom(weaponsArray)
+    let room = selectRandom(roomsArray)
+    return {suspect, weapon, room}
+}
 
 
 // ITERATION 3
 
-function revealMystery() {}
+let envelopeMystery = pickMystery()
+
+function revealMystery(envelopeMystery) {
+    return "`${envelopeMystery[0].firstName} ${envelope.Mystery[0].lastName} killed Mr. Boddy using the ${envelopeMystery[1].name} in the ${envelopeMystery[2].name}!`"
+}
```

Comment:
Phe nessa última iteração, você não vai escolher um index do envelopeMystery. Tem alguns erros na sua linha 144, vamos lá...
O return está com dois tipos de aspas, não usamos assim, ou dará erro. Usamos o literals, que são as crases, quando vamos puxar informações para esse final. 
Na iteração 2, vc nomeou variáveis (suspect, weapon, room) e vc vai usar isso no retorno da função da iteração 3 que vai revelar o misterio. 
Seu retorno da iteração 3 será assim: 
return `${envelope.suspect.firstName} ${envelope.suspect.lastName} killed Mr. Boddy using the ${envelope.weapon.name} in the ${envelope.room.name}!`
dá uma comparada pra vc ver, vê se vc entende e me avisa qualquer coisa :)